### PR TITLE
Changing mattermost-auth method to work based on shared database access

### PR DESCRIPTION
**Pull request by jespino**

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -11,14 +11,10 @@
...
 	"session_expire_time": 2592000,
 	"session_refresh_time": 18000,
 	"localOnly": false,
 	"enableLocalMode": true,
 	"localModeSocketLocation": "/var/tmp/focalboard_local.socket",
-	"authMode": "native",
-	"mattermostURL": "",
-	"mattermostClientID": "",
-	"mattermostClientSecret": ""
+	"authMode": "mattermost"
```
Comment (jespino) [inline config.json:19]: This needs to be "native" by default

File: mattermost-plugin/README.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
+# Plugin Starter Template [![CircleCI branch](https://img.shields.io/circleci/project/github/mattermost/mattermost-plugin-starter-template/master.svg)](https://circleci.com/gh/mattermost/mattermost-plugin-starter-template)
```
Comment (jespino) [inline mattermost-plugin/README.md:1]: Probably we should update this README.

File: mattermost-plugin/plugin.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
+    "id": "focalboard",
+    "name": "Focalboard",
+    "description": "This provides focalboard integration with mattermost.",
+    "homepage_url": "https://github.com/mattermost/mattermost-plugin-focalboard",
+    "support_url": "https://github.com/mattermost/mattermost-plugin-focalboard/issues",
+    "release_notes_url": "https://github.com/mattermost/mattermost-plugin-focalboard/releases/tag/v0.1.0",
+    "icon_path": "assets/starter-template-icon.svg",
+    "version": "0.1.0",
```
Comment (jespino) [inline mattermost-plugin/plugin.json:9]: Change the version to 0.2.0

File: mattermost-plugin/server/imports.go [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package main
```
Comment (jespino) [inline mattermost-plugin/server/imports.go:1]: Remove this file

File: server/api/api.go [hunk 5]
```diff
@@ -138,14 +133,21 @@ func (a *API) getContainerAllowingReadTokenForBlock(r *http.Request, blockID str
...
+		workspaceID := vars["workspaceID"]
+
 		container := store.Container{
-			WorkspaceID: "0",
+			WorkspaceID: workspaceID,
 		}
 
-		// Has session
-		if session != nil {
+		if workspaceID == "0" {
```
Comment (jespino) [inline server/api/api.go:145]: verify if this is needed.

File: server/web/webserver.go [hunk 6]
```diff
@@ -94,6 +94,10 @@ func (ws *Server) registerRoutes() {
 // Start runs the web server and start listening for charsetnnections.
 func (ws *Server) Start() {
 	ws.registerRoutes()
+	if ws.port == 0 {
```
Comment (jespino) [inline server/web/webserver.go:97]: We can change it to -1

File: server/ws/websockets.go [hunk 7]
```diff
@@ -99,9 +103,11 @@ func (ws *Server) handleWebSocketOnChange(w http.ResponseWriter, r *http.Request
 		client.Close()
 	}()
 
+	userID := r.Header.Get("Mattermost-User-Id")
```
Comment (jespino) [inline server/ws/websockets.go:106]: Also we need to check if the authentication method is mattermost, if not we can end up with a security hole here.

File: server/api/api.go (hunk 5 above)
Comment (jespino) [reply server/api/api.go:145]: This is for allowing the users to get the templates from the root workspace.

File: mattermost-plugin/webapp/src/index.tsx [hunk 8]
```diff
@@ -36,13 +36,13 @@ export default class Plugin {
         this.registry = registry;
         this.channelHeaderButtonId = registry.registerChannelHeaderButtonAction(focalboardIcon, () => {
             const currentChannel = store.getState().entities.channels.currentChannelId;
-            window.open(`${window.location.origin}/plugins/focalboard/workspace/${currentChannel}`)
+            window.open(`${window.location.origin}/plugins/focalboard/workspace/${currentChannel}`);
```
Comment (jespino) [inline mattermost-plugin/webapp/src/index.tsx:39]: We dont use `;`

Comment (mgdelacroix) [reply mattermost-plugin/webapp/src/index.tsx:39]: I think I added them because of a linter's complain, but will remove them before the merge

**Pull request merged**